I'm lazy, so let good old devfs create and remove the /dev/fuse
device node for us.

diff --git a/fs/fuse/inode.c b/fs/fuse/inode.c
index e69a546..416de4c 100644
--- a/fs/fuse/inode.c
+++ b/fs/fuse/inode.c
@@ -17,6 +17,8 @@
 #include <linux/module.h>
 #include <linux/parser.h>
 #include <linux/statfs.h>
+#include <linux/major.h>
+#include <linux/devfs_fs_kernel.h>
 
 MODULE_AUTHOR("Miklos Szeredi <miklos@szeredi.hu>");
 MODULE_DESCRIPTION("Filesystem in Userspace");
@@ -571,6 +573,7 @@ static int __init fuse_init(void)
 	if (res)
 		goto err_fs_cleanup;
 
+	devfs_mk_cdev(MKDEV(MISC_MAJOR, FUSE_MINOR), S_IFCHR | S_IRUGO | S_IWUGO, "fuse");
 	return 0;
 
  err_fs_cleanup:
@@ -583,6 +586,7 @@ static void __exit fuse_exit(void)
 {
 	printk(KERN_DEBUG "fuse exit\n");
 
+	devfs_remove("fuse");
 	fuse_fs_cleanup();
 	fuse_dev_cleanup();
 }
